drivers/video/gc2145.c

```diff
@@ -972,8 +972,7 @@ static uint8_t gc2145_check_connection(const struct device *dev)
 	}
 
 	if ((reg_ver_val != GC2145_REV_VAL) || (reg_pid_val != GC2145_PID_VAL)) {
-		LOG_ERR("failed to detect GC2145 pid: 0x%x rev: 0x%x", reg_pid_val, reg_ver_val);
-		return -EIO;
+		LOG_WRN("Unexpected GC2145 pid: 0x%x and rev: 0x%x", reg_pid_val, reg_ver_val);
```

Comment:
I suppose this is some issue you faced where the sensor returns something unexpected?
Is it consistent i.e. always the same ID for a given board or does it come with a random number?
Or is it that you tried with different boards and always had a different ID?
The answer to this would be valuable as a comment for the next folk trying to debug the sensor.
Thank you!

Reply:
I'm wondering how to do that, because whay I found is, the pid are always the same, but the revision can change, I think it would be better do push a warning here, and add to the binding a non-required id + revision pair properties.

Reply:
Interesting option: then if in a product a wrong/unexpected GalaxyCore part gets delivered, then it would be possible to immediately discover it through this error/warning

An alternative would be to check (`LOG_ERR(); return ret;`) the PID and only warn if the Rev ID changes.